### DISCUSSION TITLE
Set the IDFA identifier to no

### DISCUSSION THREAD

**Discussion by ederluca** [fastlane/fastlane#16865] [Q&A]
Good evening

When sending an update to my app I notice that the identifier parameter (IDFA) is going blank and in my case I need to set it to 'no' how do I send it automatically? What I did was put the following parameter but it still isn't working:

submission_information: {
         add_id_info_uses_idfa: false,
},

**Comment by joshdholtz** [comment]
@ederluca Hey 👋 What you are sending _should_ be correct 🤔  Can you execute your _fastlane_ with the `--verbose` flag appended on to it and past the output here? There verbose flag should should print some more info about what its trying to send for submission information and IDFA.

**Reply by joshdholtz** [reply]
@ederluca Follow up question! Do you have `skip_binary_upload` set to `false`? `submission_information` only gets sent when a binary is being uploaded.

**Reply by schunlee** [reply]
@joshdholtz Excuse me, I have a question to add here, does `submission_information` rely on `submit_for_review`, the value must be true? If set false, sounds like IDFA not changes.

**Reply by rogerluan** [reply]
@schunlee I haven't checked the code to answer to you, but I have also noticed this behavior myself. That's probably how it works (although it's not how I'd expect it to work either 😥 )

EDIT: Just checked the code and confirmed that that's how it works. It would require quite of a refactor to change this behavior from what I can see 😬

**Reply by schunlee** [reply]
@rogerluan thanks.

1. I try to use fastlane deliver's `submission_information`, and find it not like other parameters, if I set `submit_for_review` false, excepted function(content rights or IDFA) not works, sometimes I just want to change app configurations, after human check, then submit for review (I not try --- set `submit_for_review` true).

2. and I try to use spaceship directly, my code as below. After submission.complete method executed, I find IDFA changed successfully, but my app under apple's review ... ...
`require 'spaceship'`
`Spaceship::Tunes.login($FASTLANE_USER, $FASTLANE_PASSWORD)`
`app = Spaceship::Application.find(ENV['APP_IDENTIFIER'])`
`submission = app.create_submission`
`submission.add_id_info_uses_idfa = false`
`submission.complete!`

**Comment by randyvarela5** [comment]
@ederluca can you try removing the `,` after false?  I use this same flag and it works for me.

**Reply by rogerluan** [reply]
I think this will solve your issue ☝️ trailing commas are fine in Swift, but not in ruby IIRC (I can be wrong though) - and looks like you're running _fastlane_ in ruby 😊
